pulser-core/pulser/pulse.py

```diff
@@ -263,6 +265,26 @@ def __repr__(self) -> str:
             + f"post_phase_shift={self.post_phase_shift:.3g})"
         )
 
+    def __eq__(self, other: Any) -> bool:
+        if type(other) is not type(self):
+            return False
+
+        def check_phase_eq(phase1: float, phase2: float) -> np.bool_:
+            # Comparing with an offset ensures we don't fail just because
+            # we are very close to the wraping point
+            return np.isclose(phase1, phase2, atol=1e-6) or np.isclose(
+                (phase1 + 1) % (2 * np.pi),
+                (phase2 + 1) % (2 * np.pi),
+                atol=PHASE_PRECISION,
+            )
+
+        return bool(
+            self.amplitude == other.amplitude
+            and self.detuning == other.detuning
```

Comment:
Should they be equal or isclose with an absolute precision ?

Reply:
The waveform comparison already has some tolerance built-in

Reply:
Ah yes thanks for reminding me :)